services/cirrus/cirrus.service.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+const Joi = require('joi')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  message: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('unknown'))
+    .required(),
+}).required()
+
+module.exports = class Cirrus extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'cirrus',
+      pattern: '/github/:user/:repo',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Cirrus CI - Base Branch Build Status',
+        namedParams: { user: 'flutter', repo: 'flutter' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Branch Build Status',
+        pattern: '/github/:user/:repo?branch=:branch',
+        namedParams: { user: 'flutter', repo: 'flutter', branch: 'master' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Task Build Status',
+        pattern: '/github/:user/:repo?task=:task',
+        namedParams: { user: 'flutter', repo: 'flutter', task: 'analyze' },
+        staticPreview: this.render({ subject: 'analyze', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Task and Script Build Status',
+        pattern: '/github/:user/:repo?task=:task&script=:script',
+        namedParams: {
+          user: 'flutter',
+          repo: 'flutter',
+          task: 'analyze',
+          script: 'test',
+        },
+        staticPreview: this.render({ subject: 'test', status: 'passing' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { color: 'success', label: 'Cirrus CI' }
+  }
+
+  static render({ subject, status }) {
+    return renderBuildStatusBadge({ label: subject, status })
+  }
+
+  async handle({ userRepo, branch, script }) {
```

Comment:
Two objects are passed into the handle function based on the route definition, the first object is the route params and second is the query params. What I suspect you'll want here:

```suggestion
  async handle({ user, repo, branch}, { script, task }) {
```